server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchFieldsContext.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.fetch.subphase;
+
+import java.util.List;
+
+/**
+ * The context needed to retrieve fields.
+ */
+public class FetchFieldsContext {
```

Comment:
Do you see this ever being more than `List<FieldConfig>` or something like that?

Reply:
I'm wondering if it makes sense to have this class at all if it'll never end up holding more than a single list.

Reply:
I don't have immediate plans to add more information to the context. But I think it's worth having a dedicated context class for consistency with `FetchDocValuesContext` and `ScriptFieldsContext`.